app/src/main/kotlin/com/adesso/movee/scene/moviedetail/MovieDetailViewModel.kt

```diff
@@ -24,13 +24,8 @@ class MovieDetailViewModel @Inject constructor(
 
     fun fetchMovieDetails(id: Long) {
         if (_movieDetails.value == null) {
-            bgScope.launch {
-                val movieDetailResult =
-                    fetchMovieDetailUseCase.run(FetchMovieDetailUseCase.Params(id))
-
-                onUIThread {
-                    movieDetailResult.either(::handleFailure, ::postMovieDetail)
-                }
+            fetchMovieDetailUseCase.invoke(viewModelScope, FetchMovieDetailUseCase.Params(id)) {
+                it.either(::handleFailure, ::postMovieDetail)
```

Comment:
postMovieDetail() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postMovieDetail()